libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/HttpRequest.cs

```diff
@@ -194,16 +194,27 @@ public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc,
             // Single command running with a copy of the original data
             client.DefaultRequestHeaders.Clear();
 
-            JToken instanceBody = null;
+            object instanceBody = null;
+            string instanceBodyErr = null;
+
             if (this.Body != null)
             {
-                var (body, err) = this.Body.TryGetValue(dc.State);
-                if (err != null)
+                if (this.Body.Value is JObject)
```

Comment:
Do we need special cases for Dictionary<> and for dynamic in addition to JObject? If the state has those types, it wouldn't really fit any of these 2 cases. Not 100% sure that it is possible to have such a value in a declarative environment, but we could in a code only environment. @tomlm Could we leverage `ObjectPath` here to avoid doing a switch and potentially missing cases?